### ISSUE TITLE
Make more internet noise (tweet / fb / share)

### ISSUE THREAD

**Issue by jenlampton** [backdrop-ops/backdropcms.org#133]
It would be good to have someone manage the twitter account and tweet any articles posted about Backdrop, or anything relevant to our audience. 

We need a volunteer to be our social-media expert and handle these things for us. please enquire if interested: info@backdropcms.org

(Recommended by @Gormartsen in today's meeting)

TODOs - now:
1) create a blog on backdropcms.org (see https://github.com/backdrop-ops/backdropcms.org/issues/110)
2) post into that blog
3) share blog posts on social media
4) share lots of other relevant things on social media too

**Comment by docwilmot** [comment]
Great suggestion. I nominate @jenlampton and @sutibun.

**Comment by Gormartsen** [comment]
I would recommend to share all tweets and fb post somewhere in gitter or irc. So all active users can like and retweet them. 
It will help to spread info all around.

**Comment by sutibun** [comment]
\*raises hand\*

The monkey can help with this. It'll let dev work on dev stuff. Are there any guidelines of what I shouldn't say though?

> I would recommend to share all tweets and fb post somewhere in gitter or irc. So all active users can like and retweet them. 

Gitter? \*scratches head\* what's that? IRC... ah, monkey knows what that is... eh, seems a lil' geeky ^^; Haven't used it for a long time.

Do I need to post on FB too? If so, I will need to activate and authorize account since FB locked me out for security purposes because someone tried to break in. Never really used FB so didn't bother all this time. Also happens I used to help post stuff on another non-profit's FB page. This wouldn't be different.

A suggestion though don't know how feasible. I personally used [Buffer](https://buffer.com/faq/top-10-faq/#what-does-buffer-do) and the great thing about it is it allows you to broadcast one message to all your connected social profiles. I used to use a personal account and worked out well. Just checked it out and they [have a non-profit discount](https://buffer.com/faq/plans-pricing/#how-much-does-each-plan-cost) Not pushing for it... just in case.

**Comment by Gormartsen** [comment]
I can recommend to use  https://feedburner.google.com

Basically you can create, let's say a blog with RSS feed and configure  https://feedburner.google.com to cross post all entries from blog to FB and TW in realtime.

Usually it takes up to 20 minutes to get post shared on TW and FB.

This way you need access to post blog entries and TW, FB admin can configure  https://feedburner.google.com to do crossposting.

**Comment by sutibun** [comment]
Ah, that's new to me. Not sure if I fully understand. So, all I have to do is write the blog posts and Feedburner will publish them automatically to your FB + TW pages?

Where would I create the blog posts then? Bd doesn't have a blog.

**Comment by Gormartsen** [comment]
Blog should arrive soon, if I understood it properly from last meetup.

**Comment by sutibun** [comment]
Ah kk. Monkey did volunteer before to write blog posts https://github.com/backdrop-ops/backdropcms.org/issues/110 but nothing came from it ( ; ;)

**Comment by klonos** [comment]
Sorry to interrupt any conversation here. I personally never had (or ever will have - but hate saying "never") a facebook or twitter account, so I cannot help with these. ...but weren't there some Drupal modules that (re)posted specified nodes of a site to these social media platforms?

**Comment by sutibun** [comment]
\\(?_?)/ 

the monkey a Bd user... not Drupal user. hehe

uhm.. don't really know. Think tried some in the past but never could figure them out.

**Comment by klonos** [comment]
...not to sidetrack our efforts here, but we could start porting these in parallel and then use them (it would cut down in the manual work required I believe):

[Twitter](https://www.drupal.org/project/twitter):

> This module provides API integration with the Twitter microblogging service. Out of the box, it allows users to:

> - Associate one or more Twitter accounts with their Drupal user account.
> - List tweets in different ways thanks to Views.
> - Post to their own Twitter account or a site-wide Twitter account whenever they create new content or based on Drupal Actions or Rules.
> ...

Now, I realize that we haven't ported Rules yet, but Views is in core. Seems to me all we'd need (once we have a working port of this module that is) is a "Backdrop CMS" user account in backdropcms.org and access to it to whomever we choose to post blogs that are then tweeted "automagically". Right? Anyone with experience with this module in actual production?

**Comment by sutibun** [comment]
you mean the modules that are hard to use? \*shakes head\*

**Comment by Gormartsen** [comment]
Recently I was solving similar issue for drupal.ru - crossposting to TW and FB.
TW and FB api integration modules... make me feel sick. There is tone code to just do simple things.

That's why I used  https://feedburner.google.com service. It just take rss for input and automatically post all updates to social networks.

**Comment by klonos** [comment]
Don't worry Monkey, someone else will set them up for you. If these work the way I think they do (someone with experience please shed some light), then all you'd have to do is to create a piece of content there and then something like check a "publish to facebook" and/or "tweet on tweeter" checkbox (once you are given an account with proper permissions on backdropcms.org).

**Comment by sutibun** [comment]
Yeah! \*puts arm around gomar\* Monkey knew monkey wasn't the only one who thought those module are complex.

The way you explain it makes it sound easy and common sense. Try setting that up... but guess if someone else does it... monkey happy to use it.

**Comment by klonos** [comment]
If the already available modules are too complex, then perhaps instead of porting what we really need is two new, minimal "Tweeter Publish" and "Facebook Publish" modules that simply utilize the APIs that these platforms offer and all they do is to expose the "publish to facebook" and/or "tweet on tweeter" checkboxes to selected content types. Now, all we need is module developers to code these. Simple! :neutral_face:

Now, as I said, I realize this might take long. So this is not an action plan. Just a parallel todo side-project. I'm sure that if we are in need of such a solution, then so are others out there. So the developed modules will be used by other Backdrop websites. They won't be something custom-made intended for our own use :wink:

**Comment by Gormartsen** [comment]
@klonos :+1:  and include them into core then! and integrate with rules....)))

**Comment by klonos** [comment]
@Gormartsen I'm not sure whether your comment was intended to be ironic (in a nice, funny way - similar to my "Simple!" comment above), but given the fact how popular these platforms are and how important they might be SEO-wise, the suggestion for inclusion in core might not be too far-fetched. I wonder what our "friends" in WP and Joomla are handling these.

**Comment by Gormartsen** [comment]
@klonos i am not ironic at all! Mostly exited. It's really killer features to have FB and TW cross posting being integrated and attached to rules module.

**Comment by Gormartsen** [comment]
Even better. 
Let's have "Social integration" module with hook based API to connect different Social media websites: TW, LI, FB. 
(Russians developers eventually will create vk.com integration - it's most popular russian social network)

It will make BD to stand out for sure.

**Comment by sutibun** [comment]
> If the already available modules are too complex, then perhaps instead of porting what we really need is two new, minimal "Tweeter Publish" and "Facebook Publish" modules that simply utilize the APIs that these platforms offer and all they do is to expose the "publish to facebook" and/or "tweet on tweeter" checkboxes to selected content types. Now, all we need is module developers to code these. Simple! :neutral_face:

> @klonos :+1: and include them into core then! and integrate with rules....)))

monkey nominates @klonos to build it!

**Comment by klonos** [comment]
Got it :wink: It would be great marketing wise. Imagine the impact of this line: "Backdrop CMS is social out of the box!".

https://www.ostraining.com/blog/drupal/4-social-sharing-modules-drupal

**Comment by klonos** [comment]
> monkey nominates @klonos to build it!

...unfortunately, I cannot put code where my mouth is yet :disappointed:

**Comment by sutibun** [comment]
Mac computers and ipad integrate twitter and FB.

> ...unfortunately, I cannot put code where my mouth is yet :disappointed:

You've been in the Drupal community for who knows how long, you must've picked up something.

Can't nominate me... I'm a monkey.

**Comment by sutibun** [comment]
can't nominate @jenlampton either ... she's a girl... she can't code :p

**Comment by docwilmot** [comment]
Excuse me for butting in, but we're running before crawling. I would strongly recommend:
1. @Gormartsen build a blog on your own webspace, call it an (un)official BD blog
2. build the integration with feedburner
3. give @sutibun a login
4. let's start accomplishing the original aim of this post.
5. once its up @jenlampton can give keys to post to Backdrop's official pages (Twitter, FB etc)

Otherwise we may chat and plan for months before we have an 'official' BD blog. Don't talk, do!

**Comment by sutibun** [comment]
> Excuse me for butting in, but we're running before crawling. I would strongly recommend:

Eeek!

\*runs away\*

**Comment by Gormartsen** [comment]
@docwilmot I am really bad with blogging. Right now I am basically pushing BD to russian market. I just put my energy to create a wave on russian market. When I am done with it, i will step back and let people to continue this work.
2. feed burner does not need integration. It's all click and save. Once. By anyone who has access to official TW, FB and etc.

**Comment by docwilmot** [comment]
No, I meant if @sutibun cant manage the technical aspects of the feedburner business (I have no idea how to do that either) you should just build it for him, and let him start blogging. Either on your site or his, dont matter, but let's not wait for an official BD blog to start; that might be weeks. Let's just start someone blogging like next week!

**Comment by klonos** [comment]
> ...you should just build it for him, and let him start blogging. ...Let's just start someone blogging like next week!

Anything that would get the Monkey's lengthy posts out of the GitHub issue queue please :stuck_out_tongue: :smiling_imp:

**Comment by docwilmot** [comment]
Exactly; he writes interesting stuff. We need him selling Backdrop. Let's do it!

**Comment by sutibun** [comment]
Was it something the monkey said :p

**Comment by thejimbirch** [comment]
I recommend Buffer as a publishing/scheduling tool.  It can post to Twitter, Facebook (Profile, Page, Group), LinkedIn (Profile or Page), Google+ (Profile or Page), and Pinterest.  I use it personally and for MidCamp.

As for getting content into Buffer, options are:
- Manually through the web interfaces
- Apps on phones
- Integration with RSS Readers like Feedly
- RSS Feed import
- Drupal module that could be ported: https://www.drupal.org/project/bufferapp

I believe the Backdrop team also has a Mailchimp account set up.  An RSS Campaign is easy to set up and always good to distribute your newly created content via email.

**Comment by sutibun** [comment]
> I recommend Buffer as a publishing/scheduling tool...

Monkey likes it too especially since it's free for personal use. Not sure if a non-profit can use the free plan as well. That's the only reason only suggest it instead of recommend.

> I believe the Backdrop team also has a Mailchimp account set up. An RSS Campaign is easy to set up and always good to distribute your newly created content via email.

Ah, nice. Forgot about that feature. Another reason to have a blog so we can be in constant contact with those who signed up to hear more Bd stuff.

**Comment by docwilmot** [comment]
Great to have you @thejimbirch and apologies for simplistic questions, but how exactly would the team/process for this work? Can someone walk me through the setup? What we need in terms of technical setup, how the different pieces would fit, who would do the writing and how. Etc. How can I help?

**Comment by sutibun** [comment]
0. create a buffer account
1. authorize buffer to post stuff on your behalf to your social accounts (e.g. twitter, fb) Basically, link your social accounts so buffer has write permissions.
2. when you're ready, write your content (e.g. your tweet) on the web interface (or app)
3. Choose *which social accounts* you want this content to be published. i.e do you want this to appear on your FB page and twitter but not LinkedIn, OR all OR some other combo.
4. Now choose *when* you want it published: now, a specific future date/time, or on a schedule you set up before.

**Instead of going to FB, twitter, etc individually to post something, you go to this single interface where you control what, when and where to publish your content.** The biggest benefit, imo, is you get to queue up your all tweets, for example, on one day and buffer will automatically publish them for you (e.g. one per day) This will keep your social account appear active daily even though you aren't publishing it yourself.

Also, buffer has a collaborative feature where different team members can be given permission to add to the queue. So, instead of just me, other people can help to publish stuff. All this without revealing the passwords to any one of us. Buffer become a proxy. However, I've never tested this feature because I only have the free plan. I'm not sure if the discounted plan for non-profits includes this feature.

- Monkey "buffer spokesperson" :p

To summarize, basically you can

- queue up content, let buffer publish them on the schedule you set up, and make your social accounts appear active 
- possibly have more than person helping to tweet, share stuff. When one volunteer gets too busy, someone else can step in

That's really it's main appeal: centralization and collaboration.

**Comment by thejimbirch** [comment]
@sutibun As far as I know, while Backdrop is open source, it is not a 501c3 non-profit.

@docwilmot I would suggest signing up for the awesome plan, ($10/m), or jumping on an existing.  At this level you could import an RSS Feed into Buffer.  

Using Views to create a feed, you could not only include the Blog content type, but you could also include a post-to-social-media-news-type of content type.  That way you can have your content control on the site.

As for "who would do the writing and how", this can be as simple or as complicated as needed.  If you have a Workbench Moderation kind of module, you could have authors create nodes inside the site, and submit for approval when ready.  Editors/Admins can then Publish or kick back to the author for revisions.

I have also worked on many projects where content posts are written then submitted, say through Google Docs, or Basecamp, or some other offsite method, and a Webmaster-type publishes them.

**Comment by sutibun** [comment]
> I would suggest signing up for the awesome plan, ($10/m), or jumping on an existing. At this level you could import an RSS Feed into Buffer.

You can use IFTTT to get this feature with the free plan (RSS feed -> IFTTT -> Buffer)

**Comment by klonos** [comment]
> Drupal module that could be ported: https://www.drupal.org/project/bufferapp

...

> Using Views to create a feed, you could not only include the Blog content type, but you could also include a post-to-social-media-news-type of content type. That way you can have your content control on the site.

Sounds like we need to get this ported and installed in bd.org

**Comment by klonos** [comment]
...that module integrates with Rules, Token and Date. Only the last two are required and they are already in BD core!

**Comment by sutibun** [comment]
> Drupal module that could be ported: https://www.drupal.org/project/bufferapp

That module is not actually 100% necessary. The setup I explained is what I use on my Drupal site and I didn't even know that module existed. For finer controls, I simply create custom RSS feeds.

**Comment by thejimbirch** [comment]
@klonos 
The Drupal module is very nice, but is not required if you go with a Views to RSS.

**Comment by sutibun** [comment]
\*places arm around theJim\* we think alike ^^

**Comment by klonos** [comment]
Oh, OK, thanx for letting me know. Shouldn't have assumed things.

**Comment by sutibun** [comment]
Well, you at least gave me a module to check out now for my personal site ^^)b

**Comment by biolithic** [comment]
Hi,
If anyone wants to check out this ported module:
https://github.com/biolithic/backdrop_bufferapp you can.
I don't have an account, but if you have an account and will use it for
Backdrop and would like to use the module, I can clean it up and prepare it
for contrib release.
If you are going with other solutions, I will work on other things instead.

**Comment by sutibun** [comment]
Nice.

**Comment by klonos** [comment]
Thanx @biolithic :+1:

**Comment by jenlampton** [comment]
wow, you guys are fantastic :) Thank you all for jumping on this!  @sutibun yes, please help us write and promote blogs. I'm going to update the main post here with a todo for short-term and long-term based on this conversation, lmk if I got it right.

**Comment by thejimbirch** [comment]
Any backdrop-ers on Twitter, we could use a RT of this tweet:
https://twitter.com/backdropcms/status/685621966443642882

Or a social share of:
https://backdropcms.org/news/happy-new-year-backdrop-cms

**Comment by jenlampton** [comment]
We are doing a fantastic job with tweeting recently (which auto-imports into FB).  Closing this issue as resolved for now, we can re-open if necessary in the future :) High fives all around!
